core/store/migrate/migrations/0188_create_automation_upkeep_state.sql

```diff
@@ -0,0 +1,19 @@
+-- +goose Up
+
+CREATE TABLE evm_upkeep_state (
+    work_id TEXT PRIMARY KEY,
```

Comment:
What is the true underlying type of `work_id`? Is it a UUID? Hash? Arbitrary byte sequence?

Reply:
It is a hash for the evm implementation, but is represented throughout the code as a string. It's just an arbitrary identifier.

Reply:
It would be more efficient and less error prone to store an unencoded/native type if possible. Seems important for a primary key, e.g. to avoid duplicate keys like `0x1234` and `1234`.

Reply:
Please use `common.Hash` or `[32]byte` in code and `bytea` in postgres with a CHECK constraint imposing a 32 char length.